### ISSUE TITLE
[Request] More specific cause of death

### ISSUE THREAD

**Issue by rvett** [DayZMod/DayZ#711]
This request isn't a big issue, more like a "nice thing to have". 90% or so of all bodies I study have died of "blood loss". This makes it impossible to tell if somebody
- has been shot by a player with a firearm
- has been slain by a player with a melee weapon
- has been hit by a zombie
- has been eaten by a zombie
- has fallen down a cliff or building

These are things you'd probably be able to distinguish in a real life situation. It would be really neat to diversify the "blood loss" to reflect these different causes of death.

I did a bit of research. From my limited SQF programming knowledge, it would appear that what needs to be altered for this is the command `[dayz_sourceBleeding, "bled"] spawn player_death;` in [sched_medical.sqf](https://github.com/DayZMod/DayZ/blob/Development/SQF/dayz_code/system/scheduler/sched_medical.sqf#l31), written by @facoptere .

**Comment by KPanicEKS** [comment]
+1 for this: Possible verbiage below:

The player's name was "[X]" and they appear to have died from a gunshot
wound(s).     (Gunshot)
The player's name was "[X]" and they appear to have died from a gunshot
wound to the head.     (Headshot)
The player's name was "[X]" and they appear to have died from blunt force
trauma.     (Melee)
The player's name was "[X]" and they appear to have bite marks on their
body.     (Zombie)
The player's name was "[X]" and they appear to have died from falling
(Falling Death)
The player's name was "[X]" and they appear to have died from
starvation     (Starvation)
The player's name was "[X]" and they appear to have died from
dehydration     (Dehydration)
The player's name was "[X]" and they appear to have died from a crash
(Car / Air Crash) ??

**Comment by ebaydayz** [comment]
Some of those strings already exist as you can see here:
https://github.com/DayZMod/DayZ/blob/Development/SQF/dayz_code/stringtable.xml#L7558-L7629

The problem is when a player takes a body shot and bleeds out two seconds later the code currently considers that "bled" instead of gun shot. Currently the only way to have it say gun shot is via large caliber or gunshot to the head.

Fn_damageHandler could be modified to change a global variable i.e. `dayz_damageSource = "bodyShot"/"fell"/"meleeHit"/etc.` for each case and then check that global variable in sched_medical before automatically spawning player_death with reason "bled" here:
https://github.com/DayZMod/DayZ/blob/Development/SQF/dayz_code/system/scheduler/sched_medical.sqf#L30

It would need a cool down of a couple seconds, so when no other recent damage events have happened it still counts as bleeding out. 

I'll test it out and see how it works when I get to it.

**Comment by rvett** [comment]
Alright, thanks for looking into this. I'm looking forward to it!

**Comment by R4Z0R49** [comment]
We have in testing atm a new msg when killed from the infected(Zeds).

**Comment by ebaydayz** [comment]
Added here: https://github.com/DayZMod/DayZ/commit/d37526fcc62a3291fe3b7135bacb4fb172184989

I tested and confirmed zombie hit, falling, eject from vehicle and bled out are working as intended now. We can test the gunshot, melee and run over messages once this is loaded on the dev server. I also fixed the message to say her/she instead of his/he when studying a female body.

Starvation, dehydration and infection messages were already working as intended. You just don't see them much because players rarely die from them.
